pghoard/wal.py

```diff
@@ -163,7 +214,8 @@ def verify_wal(*, wal_name, fileobj=None, filepath=None):
         fmt = "WAL file {name!r} verification failed: {ex.__class__.__name__}: {ex}"
         raise ValueError(fmt.format(name=source_name, ex=ex))
 
-    expected_lsn = lsn_from_name(wal_name)
-    if hdr.lsn != expected_lsn:
+    expected_lsn = LSN.from_walfile_name(wal_name, server_version=hdr.version)
+    # Only compare the LSN value, do not pay attention to timelines here.
+    if hdr.lsn.lsn != expected_lsn.lsn:
         fmt = "Expected LSN {lsn!r} in WAL file {name!r}; found {found!r}"
-        raise LsnMismatchError(fmt.format(lsn=expected_lsn, name=source_name, found=hdr.lsn))
+        raise LsnMismatchError(fmt.format(lsn=str(expected_lsn), name=source_name, found=str(hdr.lsn)))
```

Comment:
change this in the fmt string? Eg. `fmt = "Expected LSN {lsn!s} in WAL file {name!r}; found {found!s}"`, or are the quotes important?

Reply:
The quotes are not important, but make the error message more readable I think. At least I guess that's why they were introduced, and I didn't see the point in changing the error message format.